src/main/kotlin/net/maizegenetics/phgv2/api/HaplotypeGraph.kt

```diff
@@ -0,0 +1,223 @@
+package net.maizegenetics.phgv2.api
+
+import htsjdk.variant.variantcontext.VariantContext
+import htsjdk.variant.vcf.VCFFileReader
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.Channel
+import net.maizegenetics.phgv2.utils.AltHeaderMetaData
+import net.maizegenetics.phgv2.utils.parseALTHeader
+import org.apache.logging.log4j.LogManager
+import java.io.File
+import java.util.*
+
+class HaplotypeGraph(hvcfFiles: List<String>) {
+
+    private val myLogger = LogManager.getLogger(HaplotypeGraph::class.java)
+
+    // Map<sampleName, sampleId>
+    private lateinit var sampleNameToIdMap: Map<String, Int>
+    private lateinit var sampleNames: Array<String>
+
+    // lookup sequence checksum index
+    // lookup[refRangeId][sampleId]
+    private lateinit var lookup: Array<Array<UByte>>
+
+    // seqHash[refRangeId][lookup: UByte]
+    // jagged array because different number of haplotypes for each refRange
+    private lateinit var seqHash: Array<Array<String>>
+
+    // Map<ReferenceRange, refRangeId>
+    private lateinit var refRangeMap: SortedMap<ReferenceRange, Int>
+
+    // Map<ID (checksum), AltHeaderMetaData>
+    private val altHeaderMap: MutableMap<String, AltHeaderMetaData> = mutableMapOf()
+
+    private val processingFiles = Channel<Deferred<Job>>(100)
+    private val processingChannel = Channel<RangeInfo>(10)
```

Comment:
The limits on these variables (100, 10) are what determine the parameters for multithreading, ie the coroutine limits?

Reply:
It limits the size of the Channel which can indirectly limit the multithreading